symfit/core/fit.py

```diff
@@ -356,32 +356,56 @@ class BaseNumericalModel(BaseModel):
     ABC for Numerical Models. These are models whose components are generic
     python callables.
     """
-    def __init__(self, model, independent_vars, params):
+    @keywordonly(connectivity_mapping=None)
+    def __init__(self, model, independent_vars=None, params=None, **kwargs):
         """
         :param model: dict of ``callable``, where dependent variables are the
             keys. If instead of a dict a (sequence of) ``callable`` is provided,
             it will be turned into a dict automatically.
-        :param independent_vars: The independent variables of the model.
+        :param independent_vars: The independent variables of the  model.
+            (Deprecated, use ``connectivity_mapping`` instead.)
         :param params: The parameters of the model.
-        """
-        self.independent_vars = sorted(independent_vars, key=str)
-        self.params = sorted(params, key=str)
+            (Deprecated, use ``connectivity_mapping`` instead.)
+        :param connectivity_mapping: Mapping indicating the dependencies of
+            every variable in the model. For example, a model_dict
+            {y: a * x + b} has a connectivity_mapping {y: {x, a, b}}. Note that
+            the values of this dict have to be sets.
+        """
+        connectivity_mapping = kwargs.pop('connectivity_mapping')
+        if connectivity_mapping is None and \
+                independent_vars is not None and params is not None:
+            # Make model into a mapping if needed.
+            if not isinstance(model, Mapping):
```

Comment:
Maybe my complaint about what deals with user-input where is also valid here. If you want to use this directly, you should provide the correct object. Only `Fit` gets to be clever.